03-streams/01-limit-size-stream/LimitSizeStream.js

```diff
@@ -4,10 +4,20 @@ const LimitExceededError = require('./LimitExceededError');
 class LimitSizeStream extends stream.Transform {
   constructor(options) {
     super(options);
+    this.limit = options.limit;
+    this.encoding = options.encoding;
+    this.bytesPassed = 0;
   }
 
-  _transform(chunk, encoding, callback) {
+  _transform(chunk, _encoding, callback) {
+    const chunkSize = Buffer.byteLength(chunk);
 
+    if (this.bytesPassed + chunkSize <= this.limit) {
+      callback(null, chunk);
+      this.bytesPassed += chunkSize;
+    } else {
+      callback(new LimitExceededError(), chunk);
```

Comment:
we do not need to pass `chunk` here as a 2nd argument, 'cause it'll lead to broken behavior. consider the following example:

```
const s = new LimitSizeStream({limit: 4}); // 4 bytes

s.write('lala'); // pass 4 bytes, no error
s.write('lala'); // pass additional 4 bytes - error, but 2nd chunk will also be passed meaning that in total stream will pass 8 bytes
```

Reply:
Got it, thanks 👍 
Will fix.

Reply:
Fixed.